### PR TITLE
[#1891] improvement(dev): dockerfile remove cache after installing packages

### PR DESCRIPTION
**Pull request by zhuzilong2013**
### What changes were proposed in this pull request?

Remove cache after installing packages.

### Why are the changes needed?

Fix: #1891

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

By hand

### CODE REVIEW

**Review by justinmclean — DISMISSED** [review]
LGTM

**Comment by yuqi1129** [comment]
@xunliu 
Can you assist in verifying this PR?

File: dev/docker/hive/Dockerfile [hunk 1]
```diff
@@ -151,6 +151,10 @@ RUN chown -R datastrato:hadoop /home/datastrato
 # removed install packages
 RUN rm -rf /tmp/packages
 
+################################################################################
+# removed install cache
+RUN rm -rf /var/cache/apt/*
```
Comment (xunliu) [inline dev/docker/hive/Dockerfile:156]: hi @zhuzilong2013 Thank you for your contribution.

I think it's best to use the `apt clean` command to remove the cached files, but this needs to be tested.
https://askubuntu.com/questions/1350375/can-i-delete-var-cache-apt-archives

Comment (zhuzilong2013) [reply dev/docker/hive/Dockerfile:156]: Thanks for your comment. It's my mistake. I should use `rm -rf /var/lib/apt/list/*` actually.
In https://docs.docker.com/develop/develop-images/instructions/#run:
> Official Debian and Ubuntu images automatically run apt-get clean, so explicit invocation is not required.

I will fix and test it. I also want to move 'mysql-sever' to basic tools, what do you think?

**Comment by yuqi1129** [comment]
@zhuzilong2013 
Is it ready to be reviewed again?

**Comment by zhuzilong2013** [comment]
@yuqi1129 Please review it：）

**Comment by yuqi1129** [comment]
> @yuqi1129 Please review it：）

I'm okay with the changes, @jerryshao do you have any further comments?

**Comment by qqqttt123** [comment]
@xunliu @yuqi1129 Gently ping.

**Review by xunliu — COMMENTED** [review]
hi @zhuzilong2013 Thank you for your update.
Have you tested to correctly create the Docker image using your development branch?

**Comment by zhuzilong2013** [comment]
hello @xunliu , I found the mysql silent installation failed because I moved 'mysql-sever' to basic tools. I added a new ARG to resolve it.
I build image on my M1 MacBook: 
截屏2024-02-20 18 02 52
Please review it.

File: dev/docker/hive/Dockerfile [hunk 2]
```diff
@@ -9,6 +9,7 @@ LABEL maintainer="support@datastrato.com"
 ARG HADOOP_PACKAGE_NAME
 ARG HIVE_PACKAGE_NAME
 ARG JDBC_DIVER_PACKAGE_NAME
+ARG DEBIAN_FRONTEND=noninteractive
```
Comment (xunliu) [inline dev/docker/hive/Dockerfile:12]: Where use this `DEBIAN_FRONTEND` arg? I didn't find it anywhere.

Comment (zhuzilong2013) [reply dev/docker/hive/Dockerfile:12]: I added it because I moved 'mysql-sever' to basic tools.

Comment (xunliu) [reply dev/docker/hive/Dockerfile:12]: OK, I found `DEBIAN_FRONTEND` it in the
https://askubuntu.com/questions/972516/debian-frontend-environment-variable

**Review by xunliu — APPROVED** [review]
LGTM

**Comment by mchades** [comment]
Has this PR been tested with IT? @xunliu @zhuzilong2013

**Pull request merged**